shuup/discounts/admin/views/_active_list.py

```diff
@@ -0,0 +1,88 @@
+    get_locally_formatted_datetime
+)
+
+
+class DiscountListView(PicotableListView):
+    model = Discount
+    url_identifier = "discounts"
+
+    default_columns = [
+        Column(
+            "name", _("Discount Name"), sort_field="name", display="name",
+            filter_config=TextFilter(filter_field="name", placeholder=_("Filter by name..."))
+        ),
+        Column(
+            "product__translations__name", _("Product"), display="product",
+            filter_config=TextFilter(filter_field="product__translations__name", placeholder=_("Filter by product..."))
+        ),
+        Column(
+            "category", _("Category"), display="category",
+            filter_config=TextFilter(
+                filter_field="category__translations__name", placeholder=_("Filter by category..."))
+        ),
+        Column(
+            "contact_group", _("Contact Group"), display="contact_group",
+            filter_config=TextFilter(
+                filter_field="contact_group__translations__name", placeholder=_("Filter by contact group..."))
+        ),
+        Column(
+            "contact", _("Contact"), display="contact",
+            filter_config=TextFilter(filter_field="contact__translations__name", placeholder=_("Filter by contact..."))
+        ),
+        Column(
+            "coupon_code", _("Coupon code"), display="coupon_code",
+            filter_config=TextFilter(filter_field="coupon_code__code", placeholder=_("Filter by coupon code..."))
+        ),
+        Column("discount_effect", _("Effect"), display="get_discount_effect"),
+        Column(
+            "end_datetime", _("End Date and Time"), display="format_end_datetime", filter_config=DateRangeFilter()
+        )
+    ]
+
+    mass_actions = [
+        "shuup.discounts.admin.mass_actions:ArchiveMassAction"
+    ]
+
+    def get_discount_effect(self, instance):
+        if not (instance.discount_amount_value or instance.discounted_price_value or instance.discount_percentage):
+            return "-"
+
+        effects = []
+        shop = get_shop(self.request)
+        if instance.discount_amount_value:
+            effects.append(
+                "- %s" % format_money(shop.create_price(instance.discount_amount_value))
+                if shop else format_number(instance.discount_amount_value))
+
+        if instance.discounted_price_value:
+            effects.append(
+                format_money(shop.create_price(instance.discounted_price_value))
+                if shop else format_number(instance.discounted_price_value))
+
+        if instance.discount_percentage:
+            effects.append(format_percent(instance.discount_percentage))
+
+        return ','.join(effects)
+
+    def format_end_datetime(self, instance, *args, **kwargs):
+        return get_locally_formatted_datetime(instance.end_datetime) if instance.end_datetime else ""
+
+    def get_queryset(self):
+        return Discount.objects.active(get_shop(self.request))
```

Comment:
The reason why I was not seeing the campaign I've just created is because the `active` here.. the dates were not in the range neither the active checkbox was selected.. does this make sense?

Reply:
Yeah that is missing shop. And yes it is on purpose that active and not active discounts is shown in separate list to avoid confusion.